eda/src/autogluon/eda/visualization/layouts.py

```diff
@@ -78,14 +86,70 @@ class MarkdownSectionComponent(AbstractVisualization, JupyterMixin):
     Render provided string as a Markdown cell.
     See `Jupyter Markdown cell <https://jupyter-notebook.readthedocs.io/en/stable/examples/Notebook/Working%20With%20Markdown%20Cells.html>`_
     documentation for details.
+
+    Parameters
+    ----------
+    markdown: str
+        markdown text to render
+    condition_fn: Optional[Callable], default = None
+        if specified, call the provided function with `state` arg. The function expected to return bool value
+    namespace: Optional[str], default = None
+        namespace to use; can be nested like `ns_a.ns_b.ns_c`
+
     """
 
-    def __init__(self, markdown: str, namespace: Optional[str] = None, **kwargs) -> None:
+    def __init__(
+        self, markdown: str, condition_fn: Optional[Callable] = None, namespace: Optional[str] = None, **kwargs
+    ) -> None:
+        """
+
+        Parameters
+        ----------
+        markdown
+        condition_fn
+        namespace
+        kwargs
+        """
         super().__init__(namespace, **kwargs)
         self.markdown = markdown
+        self.condition_fn = condition_fn
 
     def can_handle(self, state: AnalysisState) -> bool:
-        return True
+        return True if self.condition_fn is None else self.condition_fn(state)
 
     def _render(self, state: AnalysisState) -> None:
         self.render_markdown(self.markdown)
+
+
+class PropertyRendererComponent(AbstractVisualization, JupyterMixin):
+    """
+    Render component stored in `state`'s dot-separated path to property (i.e. `a.b` results in `state.a.b`)
+
+    Parameters
+    ----------
+    property: str
+        dot-separated path to property (i.e. `a.b` results in `state.a.b`)
+    transform_fn: Optional[Callable], default = None
+        if specified, call the provided function with the object extracted from `state`'s `property`.
+        Returned transformation is then passed into render function
+    namespace: Optional[str], default = None
+        namespace to use; can be nested like `ns_a.ns_b.ns_c`
+    """
+
+    def __init__(
+        self, property: str, transform_fn: Optional[Callable] = None, namespace: Optional[str] = None, **kwargs
+    ) -> None:
+        super().__init__(namespace, **kwargs)
+        self.property = property
+        self.transform_fn = transform_fn
+
+    def can_handle(self, state: AnalysisState) -> bool:
+        return is_key_present_in_state(state, self.property)
+
+    def _render(self, state: AnalysisState) -> None:
+        obj = state
+        for p in self.property.split("."):
+            obj = obj[p]
```

Comment:
Is there a risk that properties could have the same name and so this overwrites earlier properties?

Reply:
You can think about state as a nested maps; so no duplicate keys; writing into existing key will overwrite earlier version.